packages/fuels-core/src/types/tx_status.rs

```diff
@@ -15,43 +15,91 @@ use crate::{
     types::errors::{transaction::Reason, Error, Result},
 };
 
+#[derive(Debug, Clone)]
+pub struct Success {
+    pub receipts: Vec<Receipt>,
+    pub total_fee: u64,
+    pub total_gas: u64,
+}
+
+#[derive(Debug, Clone)]
+pub struct SqueezedOut {
+    pub reason: String,
+}
+
+#[derive(Debug, Clone)]
+pub struct Revert {
+    pub reason: String,
+    pub receipts: Vec<Receipt>,
+    pub revert_id: u64,
+    pub total_fee: u64,
+    pub total_gas: u64,
+}
+
 #[derive(Debug, Clone)]
 pub enum TxStatus {
-    Success {
-        receipts: Vec<Receipt>,
-    },
+    Success(Success),
     Submitted,
-    SqueezedOut {
-        reason: String,
-    },
-    Revert {
-        receipts: Vec<Receipt>,
-        reason: String,
-        revert_id: u64,
-    },
+    SqueezedOut(SqueezedOut),
+    Revert(Revert),
 }
 
 impl TxStatus {
     pub fn check(&self, log_decoder: Option<&LogDecoder>) -> Result<()> {
         match self {
-            Self::SqueezedOut { reason } => {
+            Self::SqueezedOut(SqueezedOut { reason }) => {
                 Err(Error::Transaction(Reason::SqueezedOut(reason.clone())))
             }
-            Self::Revert {
+            Self::Revert(Revert {
                 receipts,
                 reason,
                 revert_id: id,
-            } => Self::map_revert_error(receipts, reason, *id, log_decoder),
+                ..
+            }) => Err(Self::map_revert_error(receipts, reason, *id, log_decoder)),
             _ => Ok(()),
         }
     }
 
+    pub fn take_success_checked(self, log_decoder: Option<&LogDecoder>) -> Result<Success> {
+        match self {
+            Self::SqueezedOut(SqueezedOut { reason }) => {
+                Err(Error::Transaction(Reason::SqueezedOut(reason.clone())))
+            }
+            Self::Revert(Revert {
+                receipts,
+                reason,
+                revert_id: id,
+                ..
+            }) => Err(Self::map_revert_error(&receipts, &reason, id, log_decoder)),
+            Self::Submitted => Err(Error::Transaction(Reason::SqueezedOut(
```

Comment:
`Reason::SqueezedOut` is misplaced. 

IMO the error message doesn't need to convey details about the structures are being processed internally ("cannot take x from y").
Something like "transactions was not yet included" contains just enough info.

Reply:
Good catch thanks!